kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/ResponsiveMetrics.java

```diff
@@ -48,10 +48,12 @@ public class ResponsiveMetrics implements Closeable {
   private static final Pattern GLOBAL_THREAD_REGEX = Pattern.compile(".*-(GlobalStreamThread+)");
 
   private OrderedTagsSupplier orderedTagsSupplier;
+  private final OtelMetricsService otelService;
```

Comment:
can we keep this service decoupled from the metrics collection stuff? Ideally we won't bake in the fact that we're using any specific reporter. One suggestion is to have some way to listen on `ResponsiveMetrics` being closed and register the otel reporter as a listener. And then move the otel stuff to its own package.

Reply:
I'll wrap it in a `MetricsExporterService` and move no-op to be it's own implementation of that. That way we don't add yet another series of callbacks, they're hard to keep track of.